Automattic/wp-calypso
# Checkout-composite: Stub out coupon field

#### Changes proposed in this Pull Request

![image](https://user-images.githubusercontent.com/6981253/68357491-89efed00-00e3-11ea-9ca7-d03301496674.png)


This PR stubs in the coupon field with the following functionality
- Submits when you're focused on the field and press enter/return 
- Display success/error after submitting
- Shows the apply button after you enter your first character and then hides it if you have no characters in the field.
- Has a toggle in the summary to display the field. 
- Hides the coupon field, and toggle in the summary, after submission 

Notes:
- The coupon field is always visible in the review step but requires a toggle to display it in the summary step. 
- If you submit the coupon from one field, it should hide the other field too but that's currently not the case because we need to implement adding the coupon to the line items data object first.

#### Testing instructions


* Get up and running: https://github.com/Automattic/wp-calypso/pull/37013
* Press add coupon
* Enter anything and press enter on the keyboard to see success
* Complete the steps
* Enter another coupon in the review step
* Submit to the see error
